linux_os/guide/services/ntp/service_timesyncd_configured/bash/shared.sh

```diff
@@ -5,22 +5,23 @@
 # disruption = low
 
 {{{ bash_instantiate_variables("var_multiple_time_servers") }}}
-
 IFS=',' read -r -a time_servers_array <<< "$var_multiple_time_servers"
 preferred_ntp_servers_array=("${time_servers_array[@]:0:2}")
 preferred_ntp_servers=$( echo "${preferred_ntp_servers_array[@]}"|sed -e 's/\s\+/,/g' )
 fallback_ntp_servers_array=("${time_servers_array[@]:2}")
 fallback_ntp_servers=$( echo "${fallback_ntp_servers_array[@]}"|sed -e 's/\s\+/,/g' )
 
+IFS=" " mapfile -t current_cfg_arr < <(ls -1 /etc/systemd/timesyncd.d/* 2>/dev/null)
 config_file="/etc/systemd/timesyncd.d/oscap-remedy.conf"
-current_cfg_arr=( "/etc/systemd/timesyncd.conf" )
-current_cfg_arr+=("$(ls /etc/systemd/timesyncd.d/*)")
+current_cfg_arr+=( "/etc/systemd/timesyncd.conf" )
 # Comment existing NTP FallbackNTP settings
-for current_cfg in "${current_cfg_arr[@]}"
-do
-    sed -i 's/^NTP/#&/g' "$current_cfg"
-    sed -i 's/^FallbackNTP/#&/g' "$current_cfg"
-done
+if [ ${#current_cfg_arr[@]} -ne 0 ]; then
```

Comment:
I think this condition is superfluous because the array will always have at least one element because of the assignment on line 16.

Reply:
Thanks - dropped in [669c80c](https://github.com/ComplianceAsCode/content/pull/11247/commits/669c80cd6c1d4316e3800cd779c751142a59fbd4)